packages/mds-stream/kafka/stream-consumer.ts

```diff
@@ -28,9 +29,50 @@ export interface KafkaStreamConsumerOptions {
   fromBeginning: boolean
 }
 
+const registerLivelinessHandling = (consumer: Consumer, healthStatus: HealthStatus) => {
+  let lastHeartbeat: Nullable<Timestamp> = null
+
+  const { CRASH, STOP, DISCONNECT, HEARTBEAT } = consumer.events
+
+  consumer.on(CRASH, async ({ error, groupId, restart }) => {
+    StreamLogger.error('Kafka consumer crashed', { error, groupId, restart })
+    await disconnectConsumer(consumer)
+    healthStatus.healthy = false
+    healthStatus.message = `Kafka consumer crashed: ${JSON.stringify(error)}`
+  })
+  consumer.on(STOP, async () => {
+    StreamLogger.error('The Kafka consumer has stopped')
+  })
+  consumer.on(DISCONNECT, async () => {
+    StreamLogger.error('The Kafka consumer has disconnected')
+  })
+  consumer.on(HEARTBEAT, async () => {
+    StreamLogger.debug('The Kafka consumer has sent a heartbeat to the coordinator')
+    lastHeartbeat = now()
+  })
+
+  /**
+   * If we haven't sent a heartbeat to the coordinator in over a minute, restart the consumer pod entirely
+   */
+  const livelinessHandler = async () => {
+    if (lastHeartbeat && now() - 60_000 > lastHeartbeat) {
```

Comment:
magic number should be declared by name, maybe default 60s + loaded from process.env

Reply:
100%, I was planning on making this configurable too :)

Reply:
Resolved in [0ec7375](https://github.com/lacuna-tech/mds-core/pull/991/commits/0ec73758ba3205d010e8c7704ae42dc93b69cfc6)